### PR TITLE
As a developer I can receive headers while using download_one()

### PR DESCRIPTION
**Pull request by ipanova**
closes #1179
https://pulp.plan.io/issues/1179

### CODE REVIEW

**Comment by ipanova** [comment]
I created a test that checks and proofs that download_headers event is fired much earlier than the download_succeed/fail event.

```
import datetime

from nectar import listener, request, config
from nectar.downloaders import threaded

download_config = config.DownloaderConfig()


class MyEventListener(listener.DownloadEventListener):

    def download_succeeded(self, report):
        print "Download succeeded at %s" % str(datetime.datetime.now())

    def download_failed(self, report):
        print "Download failed at %s" % str(datetime.datetime.now())

    def download_headers(self, report):
        print report.headers
        print "Headers were downloaded at %s" % str(datetime.datetime.now())

download_config = config.DownloaderConfig()

listener = MyEventListener()

downloader = threaded.HTTPThreadedDownloader(download_config, listener)

request = request.DownloadRequest('https://archive.fedoraproject.org/pub/archive/fedora/linux/core/6/i386/iso/FC-6-i386-rescuecd.iso', '/home/ipanova/fedora.iso')

report = downloader.download_one(request, events=True)
```

**Comment by ipanova** [comment]
@bmbouter  I added the docs, so it is ready for review.

File: nectar/downloaders/threaded.py [hunk 1]
```diff
@@ -215,10 +215,14 @@ def _fetch(self, request, session):
...
-                    response = session.get(
-                        request.url, headers=headers, timeout=(self.config.connect_timeout,
-                                                               self.config.read_timeout)
-                    )
+
+                    response = session.get(request.url, headers=headers,
+                                           timeout=(self.config.connect_timeout,
+                                                    self.config.read_timeout))
+
+                    report.headers = response.headers
```
Comment (bmbouter) [inline nectar/downloaders/threaded.py]: @ipanova Is there any chance that response.headers won't be available when this runs? For example say the webserver is really slow and doesn't send the headers quickly. This code will execute and could it just save a reference to response.headers, or maybe it's None? I think if you use Nectar to make a request to a pulp URL that has a `pdb` statement in it which prevents the response from being formed, then we should wait to see the `download_headers` event until after the response (including the headers) has been returned from Django.

**Comment by bmbouter** [comment]
Once this gets merged, there are still some things that need to be done. The [spec file version](https://github.com/pulp/nectar/blob/master/python-nectar.spec#L4) needs to be updated, and the new version needs to be built. I think the new version should be 1.4.0 since this is a new feature. What do you think?

Also the new version needs to be committed into pulp/pulp in the deps folder.

**Comment by bmbouter** [comment]
I tested the reproducer and I agree the headers were available much earlier than the download completed.

```
{'apptime': 'D=3042', 'content-length': '89618432', 'appserver': 'download04.phx2.fedoraproject.org', 'accept-ranges': 'bytes', 'keep-alive': 'timeout=5, max=100', 'server': 'Apache/2.4.6 (Red Hat Enterprise Linux) OpenSSL/1.0.1e-fips', 'last-modified': 'Wed, 18 Oct 2006 00:15:51 GMT', 'connection': 'Keep-Alive', 'etag': '"5577800-4200ac607bbc0"', 'date': 'Wed, 16 Sep 2015 12:55:25 GMT', 'content-type': 'application/octet-stream'}
Headers were downloaded at 2015-09-16 08:55:25.594869
Download succeeded at 2015-09-16 08:58:11.578154
```

File: nectar/downloaders/threaded.py (hunk 1 above)
Comment (ipanova) [inline nectar/downloaders/threaded.py]: I asked on the python-requests channel and developers confirmed that the response will not be delivered until the headers will be available. So we are safe here.

**Comment by ipanova** [comment]
@bmbouter I agree that the new version should be 1.4.0

File: nectar/downloaders/threaded.py (hunk 1 above)
Comment (bmbouter) [inline nectar/downloaders/threaded.py]: I have also confirmed this functionality. The `response = session.get(...)` does not return until the headers are already received by `python-requests`. My concerns are resolved.

**Comment by bmbouter** [comment]
@ipanova I'm done with my review, this work is well done! Maybe redo the commit to have `re #1179` instead of `closes #1179` so that Redmine won't put the issue to MODIFIED until the koji build is done and it's been included in downstream Pulp. I'm assigning back to you, LGTM.

**Pull request merged**